### ISSUE TITLE
Content type caching breaks for multiple spaces with same content type names

### ISSUE THREAD

**Issue by EricNelson12** [contentful/contentful.py#73]
Since `ContentTypeCache` has a static field `__CACHE__ = []`, this `__CACHE__` object is reused across clients.

`ContentTypeCache`'s methods `get` and `update_cache` are called statically instead of on different instances of `ContentTypeCache`

So if I have two spaces, and need two clients like

```python
client_for_space_1 = contentful.Client(<credentials>...,space_id = 1)
client_for_space_2 = contentful.Client(<credentials>..., space_id = 2)
```

When `update_cache` is called for the second client, it will replace the cache made by the first client.

This might go undetected if the spaces have different content type ids or content types with different field names. But if the spaces have the same content type ids, then when the cache is called like:

```python
content_type = ContentTypeCache.get(
            self.sys['content_type'].id
        )
```

It will happily return a content type which might not have the same field type. This causes errors if for example, it expects a dict, but instead gets a string type.

So either new instances of `ContentTypeCache` should be created per client, or the key to `__CACHE__` should be prefixed with the space id.

Thanks for reading!

**Comment by rubydog** [comment]
Hey @EricNelson12 thanks for reporting the issue.   

I am looking into it. I will get back to you if I need more details.

Cheers

**Comment by EricNelson12** [comment]
Thank you very much @rubydog !

**Comment by EricNelson12** [comment]
Also @rubydog I can assist with testing if that would help :smile:

**Comment by rubydog** [comment]
Hi @EricNelson12, I have started making changes in caching strategy. You can review PR #78 :)

**Comment by rubydog** [comment]
Fixed in PR #78
